desktop/test/render-visdiff.js

```diff
@@ -0,0 +1,48 @@
+import path from 'path'
+import fs from 'fs'
+import {app, BrowserWindow, ipcMain} from 'electron'
+import {resolveRoot, resolveRootAsURL} from '../resolve-root'
+import dumbMap from '../shared/more/dumb-map'
+
+const outputDir = resolveRoot('screenshots')
+if (!fs.existsSync(outputDir)) {
+  fs.mkdirSync(outputDir)
+}
+
+let win
+
+const toRender = []
+Object.keys(dumbMap).forEach(key => {
+  if (key === 'Tracker') {
+    // FIXME: Tracker dumb components aren't fully stateless yet
+    return
+  }
+  Object.keys(dumbMap[key].mocks).forEach(mockKey => {
+    toRender.push({key, mockKey})
+  })
+})
+
+function renderNext () {
+  if (!toRender.length) {
+    app.quit()
+    return
+  }
+  win.webContents.send('display', toRender.pop())
+}
+
+ipcMain.on('display-done', (ev, msg) => {
+  win.capturePage(msg.rect, img => {
+    const filenameParts = [msg.key, msg.mockKey].map(s => s.replace(' ', '_').replace(/[^\w]*/g, ''))
+    const filename = filenameParts.join('-') + '.png'
+    fs.writeFileSync(path.join(outputDir, filename), img.toPng())
+    console.log('wrote', filename)
+    renderNext()
+  })
+})
+
+app.on('ready', () => {
+  win = new BrowserWindow({show: false, width: 1000, height: 1000})
```

Comment:
🐑 would be a little nicer if `win` was set once as `const` and passed around, instead of just using one instance of `win`